core/src/test/java/org/testcontainers/containers/FixedHostPortGenericContainerTest.java

```diff
@@ -0,0 +1,27 @@
+package org.testcontainers.containers;
+
+import org.junit.Ignore;
+import org.junit.Test;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import java.net.InetSocketAddress;
+import java.net.Socket;
+
+public class FixedHostPortGenericContainerTest {
+    @Test(timeout = 10_000)
+    @Ignore // Test is disabled by default, as in general it is not reliable to run a test using fixed port number
```

Comment:
Perhaps we could find an available port on the host first, and then use that instead of `8080`?

Reply:
I don't know of a reliable way of doing this.

One could do something like this: https://github.com/airlift/airlift/blob/e739d06ae27a6de059a0b5c56b06ba8c69ae712e/jmx/src/main/java/io/airlift/jmx/NetUtils.java#L26-L37

in theory is should work, since port should not be implicitly reused for some time, unless explicitly reopened by the same OS user.
in practice, however, using this technique was proven to cause test flakiness

Do you have any advice?

Reply:
Do we really need an integration test for this?
Why not just make a unit test and check value of `getExposedPorts()` method to contain the port?

We have other (integration) tests to make sure, that exposing ports works.

Reply:
I agree with @kiview that it can be a unit test instead :)

However, if we go the integration test way, check other tests for the fixed port functionality, we have a few.